partner/app/main.py

```diff
@@ -45,20 +45,49 @@ async def read_root():
     return {"message": message}
 
 
-@app.get("/tilesp", response_model=TilesResponse)
+@app.get("/tilesp", response_model=Tiles, status_code=200)
 async def read_tilesp(
+    response: Response,
     partner: str = Query(..., example="demofeed"),
     sub1: str = Query(..., example="123456789"),
     sub2: str = Query(..., example="placement1"),
     country_code: str = Query(..., alias="country-code", example="US"),
     region_code: str = Query(..., alias="region-code", example="NY"),
     form_factor: str = Query(..., alias="form-factor", example="desktop"),
-    os_family: str = Query(..., alias="os-family", example="macOS"),
+    os_family: str = Query(..., alias="os-family", example="macos"),
     v: str = Query(..., example="1.0"),
     out: str = Query("json", example="json"),
     results: int = Query(1, example=2),
 ):
     """Endpoint for requests from Contile."""
-    tiles = TilesResponse(tiles=available_tiles[:results])
-    logger.debug(json.dumps(tiles, default=pydantic_encoder))
-    return tiles
+    # Read response information from the response.yml file
+    response_from_file = responses_from_file[form_factor][os_family]
+
+    # Read response information from the file
+    status_code = response_from_file.status_code
+    content = response_from_file.content
+    delay = response_from_file.delay
+    headers = {header.name: header.value for header in response_from_file.headers}
+
+    if delay:
+        # Add an artificual delay to the handler
+        logger.debug("response is delayed by %s seconds", delay)
+        time.sleep(delay)
```

Comment:
This being an async function, you don't want time.sleep blocking the event loop
```suggestion
        asyncio.sleep(delay)
```

Reply:
Good catch! I updated this to use `await asyncio.sleep(delay)` instead. We need to `await` to ensure the HTTP request from contile actually times out.